transcom/mymove
# B-18066-INTEGRATION

## [B-18066](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A870633)
### [E-05254](https://www13.v1host.com/USTRANSCOM38/Epic.mvc/Summary?oidToken=Epic:908764)

## Summary
 **For INTEGRATION**

This PR is for the PPM down payment packet API.

Sample PDF: 
[ppm_payment_packet-2024-03-05T13_09_53.021Z.pdf](https://github.com/transcom/mymove/files/14496670/ppm_payment_packet-2024-03-05T13_09_53.021Z.pdf)

Note: PDF will also contain both bookmarks and footer/watermark.

Internal:
- GET: /ppm-shipments/{ppmShipmentId}/payment-packet/

GHC/Office
- GET: /ppm-shipments/{ppmShipmentId}/payment-packet/

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Login into Milmove customer 
2. on board PPM and submit
3. Login into Milmove office and review move
4. Approve PPM and "move details"
5. Login into Milmove customer and start uploading documentations(see UI attachment)
6.  Add weights, pro-gear, expenses ..etc..
7. Determine ppm shipmentId. Go to customer/office swagger -- "payment-packet" and execute API.
8. Swagger will also provide a URL(not curl example). Copy and paste into browser.  This is the quickest way to test without using postman. Make sure you are logged into either customer or office.
`http://milmovelocal:3000/internal/ppm-shipments/ff084acc-77ef-4d87-8c02-aff83aa56f74/payment-packet`
9. Run in Postman if applicable

![Screenshot 2024-03-05 at 2 27 48 PM](https://github.com/transcom/mymove/assets/148127004/a8934025-9ca7-48ad-9030-911ecf52b76b)

### Backend
